### PR TITLE
[PWA-589] Page Reset on Sort

### PR DESCRIPTION
**Pull request by eug123**
## Description

If a customer changes the sort criteria they would be returned to the first 'page' of the newly sorted items.

## Related Issue

https://jira.corp.magento.com/browse/PWA-589

### Verification Stakeholders

@dhaecker 
@jimbo 

### Verification Steps

1. Go to a Category/Search results page
2. Sort by Price: High to Low
3. Go to page 2
4. Change the sort to Price: Low to High
5. See that you sent back to page 1

## Checklist

- [ ] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by devops-pwa-codebuild** [comment]
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2458.pwa-venia.com : LH Performance Expected 0.85 Actual 0.58, LH Best Practices Expected 1 Actual 0.92
https://pr-2458.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.34, LH Best Practices Expected 1 Actual 0.92
https://pr-2458.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.5, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92

File: packages/peregrine/lib/talons/SearchPage/__tests__/useSearchPage.spec.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                    total_pages: 6
+                }
+            }
+        },
+        error: null,
+        loading: false
+    };
+    const useLazyQuery = jest.fn(() => [runQuery, queryResult]);
+
+    return { runQuery, queryResult, useLazyQuery, useQuery };
```
Comment (tjwiebell) [inline packages/peregrine/lib/talons/SearchPage/__tests__/useSearchPage.spec.js:81]: Not causing any harm, but these shouldn't be in the return.

```suggestion
    return { useLazyQuery, useQuery };
```

File: packages/peregrine/lib/talons/SearchPage/__tests__/useSearchPage.spec.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        {
+            sortText: 'Best Match',
+            sortAttribute: 'relevance',
+            sortDirection: 'Changed'
+        },
+        1
+    ]
+];
+
+test.each(testCases)(
```
Comment (tjwiebell) [inline packages/peregrine/lib/talons/SearchPage/__tests__/useSearchPage.spec.js:136]: This reminds me of DataProviders in PHPUnit and I think I like it. May just have to use this pattern in the next set of tests I write, looks awesome.

File: packages/venia-ui/lib/RootComponents/Category/category.js [hunk 3]
```diff
@@ -28,6 +28,9 @@ const Category = props => {
     const sortProps = useSort();
     const [currentSort] = sortProps;
 
+    // Keep track of the sort criteria so we can tell when they change.
+    const previousSort = useRef(currentSort);
```
[the remark this answers is not in the document]
Comment (eug123) [reply packages/venia-ui/lib/RootComponents/Category/category.js:32]: used `previousSearch` example)

File: packages/peregrine/lib/talons/SearchPage/__tests__/useSearchPage.spec.js (hunk 1 above)
Comment (eug123) [reply packages/peregrine/lib/talons/SearchPage/__tests__/useSearchPage.spec.js]: updated

**Comment by dhaecker** [comment]
QA approved

**Pull request merged**